plugins/ConvertStacked/ConvertStacked.cpp

```diff
@@ -134,13 +134,41 @@ class ConvertFromStacked : public GenericVideoFilter
 {
 public:
 
-    ConvertFromStacked::ConvertFromStacked(PClip src, IScriptEnvironment* env) : GenericVideoFilter(src)
+    ConvertFromStacked::ConvertFromStacked(PClip src, int bits, IScriptEnvironment* env) : GenericVideoFilter(src)
     {
-        if (vi.IsYV12()) vi.pixel_type = VideoInfo::CS_YUV420P16;
-        else if (vi.IsYV16()) vi.pixel_type = VideoInfo::CS_YUV422P16;
-        else if (vi.IsYV24()) vi.pixel_type = VideoInfo::CS_YUV444P16;
-        else if (vi.IsY8()) vi.pixel_type = VideoInfo::CS_Y16;
-        else env->ThrowError("ConvertFromStacked: Input stacked clip must be YV12, YV16, YV24 or Y8");
+        if (bits == 10 && vi.IsYV12())
+            vi.pixel_type = VideoInfo::CS_YUV420P10;
+        else if (bits == 10 && vi.IsYV16())
+            vi.pixel_type = VideoInfo::CS_YUV422P10;
+        else if (bits == 10 && vi.IsYV24())
+            vi.pixel_type = VideoInfo::CS_YUV444P10;
+        else if (bits == 10 && vi.IsY8())
+            vi.pixel_type = VideoInfo::CS_Y10;
+        else if (bits == 12 && vi.IsYV12())
+            vi.pixel_type = VideoInfo::CS_YUV420P12;
+        else if (bits == 12 && vi.IsYV16())
+            vi.pixel_type = VideoInfo::CS_YUV422P12;
+        else if (bits == 12 && vi.IsYV24())
+            vi.pixel_type = VideoInfo::CS_YUV444P12;
+        else if (bits == 12 && vi.IsY8())
+            vi.pixel_type = VideoInfo::CS_Y12;
+        else if (bits == 14 && vi.IsYV12())
+            vi.pixel_type = VideoInfo::CS_YUV420P14;
+        else if (bits == 14 && vi.IsYV16())
+            vi.pixel_type = VideoInfo::CS_YUV422P14;
+        else if (bits == 14 && vi.IsYV24())
+            vi.pixel_type = VideoInfo::CS_YUV444P14;
+        else if (bits == 14 && vi.IsY8())
+            vi.pixel_type = VideoInfo::CS_Y14;
+        else if (bits == 16 && vi.IsYV12())
+            vi.pixel_type = VideoInfo::CS_YUV420P16;
+        else if (bits == 16 && vi.IsYV16())
+            vi.pixel_type = VideoInfo::CS_YUV422P16;
+        else if (bits == 16 && vi.IsYV24())
+            vi.pixel_type = VideoInfo::CS_YUV444P16;
+        else if (bits == 16 && vi.IsY8())
+            vi.pixel_type = VideoInfo::CS_Y16;
+        else env->ThrowError("ConvertStackedToNative: Input stacked clip must be YV12, YV16, YV24 or Y8");
```

Comment:
1.) You have an is_yuva() function. If true, the number of planes is 4.
Or call vi.ComponentCount. It will return 3 for Planar RGB and YUV, and 4 for Planar RGBA or YUVA.
2.) I can see a serious confusion on bits_per_component usage.
When you ask for 8 bit types it returns 8. For 16 bit types it returns 10-12-14-16.
In the linked ffmpeg source bits_per_component is used for yuv420 types. For 8 bit the function returns bits=8. For YV12 the result is 3x8=24. For a YV16 or YV24 it will also return 3*8=24

The "harcoded" avs 2.6 branch calls the classic "bits_per_pixel" function which is a special function that returns the _effective_ _average_ bit accupation for whole pixel. Let's see a YV12 (y-u-v) pixel. Classic bitsPerPixel returns 12 (reason: for 4 pixels = 4 bytes luma + 2 byte chroma = 6 byte / 4 pixel -> 1.5 byte / 1 pixel -> 12 bits for a pixel.)

You have to decide what that bits really means here. I can see that a stream-like packet size is calculated from "bits".
I recommend that the Classic avs bitsperPixel calculation should be used then additionally be scaled with *bits_per_component/8.

That also implies that a 10 bit avisynth format contains real 10 bit data and not scaled to 16 (?).

3.) Then later comes when rgb is checked. Planar RGB is also top bottom?

That's all for now, I will possibly off for a week from now (holiday).